packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -388,29 +385,29 @@ const findLane = (y1, y2, depth = 0) => {
  * Draw the lines between the commits. They were arrows initially.
  *
  * @param {any} svg
- * @param {any} commit1
- * @param {any} commit2
+ * @param {any} commitA
+ * @param {any} commitB
  * @param {any} allCommits
  */
-const drawArrow = (svg, commit1, commit2, allCommits) => {
-  const p1 = commitPos[commit1.id];
-  const p2 = commitPos[commit2.id];
-  const overlappingCommits = hasOverlappingCommits(commit1, commit2, allCommits);
-  // log.debug('drawArrow', p1, p2, overlappingCommits, commit1.id, commit2.id);
+const drawArrow = (svg, commitA, commitB, allCommits) => {
+  const p1 = commitPos[commitA.id];
+  const p2 = commitPos[commitB.id];
+  const overlappingCommits = hasOverlappingCommits(commitA, commitB, allCommits);
+  // log.debug('drawArrow', p1, p2, overlappingCommits, commitA.id, commitB.id);
 
   let arc = '';
   let arc2 = '';
   let radius = 0;
   let offset = 0;
-  let colorClassNum = branchPos[commit2.branch].index;
+  let colorClassNum = branchPos[commitB.branch].index;
   let lineDef;
   if (overlappingCommits) {
     arc = 'A 10 10, 0, 0, 0,';
     arc2 = 'A 10 10, 0, 0, 1,';
     radius = 10;
     offset = 10;
     // Figure out the color of the arrow,arrows going down take the color from the destination branch
-    colorClassNum = branchPos[commit2.branch].index;
+    colorClassNum = branchPos[(p1.y > p2.y ? commitA : commitB).branch].index;
```

Comment:
We need more integration tests to cover this case

Reply:
This line has now gone and the `colorClassNum` is defined within existing conditions to make sure `TB` cases are accounted for. So I think it's covered by existing integration tests.